src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -847,6 +847,13 @@ public void Run()
                 // currently fire differently from one another. 
                 foreach (NodeModel nodeModel in task.ModifiedNodes)
                 {
+                    // CodeBlockNodes are unique in that they can exhibit two types of warnings and errors - 
+                    // those that are found at precompilation (compile time errors and warnings),
+                    // and those that are found at execution time (runtime errors and warnings). In the case
+                    // of CBNs therefore, we need to retain errors and warnings from both stages, which is why we cannot 
+                    // clear them here (before execution) as this will wipe out those from the precompilation stage.
+                    if (nodeModel is CodeBlockNodeModel) continue;
```

Comment:
@mjkkirschner this is the actual fix for the bug where the warning didn't show up on the node after executing it even though the graph said `Run completed with warnings.`

Reply:
After a lot of debugging, I found actually this `nodeModel.OnNodeMessagesClearing();` here will clear node warnings include persistent warnings as well... Still need to look at why that changed in 2.13, but I feel persistent warning is another case here we should not clear after execution, otherwise they simply get lost. Do you have any memory on that end?

Reply:
@QilongTang do you mean persistent warnings that are related to obsolete nodes also need to be excluded from clearing over here? Will check that.

Reply:
It's fine @aparajit-pratap Not meant to put all bugs on you. If this PR is good to merge, we can always follow up

Reply:
Thanks for pointing this out @QilongTang, I actually think I have a better solution now that treats all persistent warning nodes consistently so I don't have to call out CBN's specifically. Will send an update after some more testing.